File: listr-svc/ListR/ListR.Common/EntityModels/Lists/ShopList.cs


using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ListR.DataLayer.EntityModels.Lists
{
    public class ShopList
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string color { get; set; } = string.Empty;

        public bool isDeleted { get; set; } = false;

        public bool isCompleted { get; set; } = false;

        public DateTime DateCreated { get; set; }

        public DateTime? DateModified { get; set; }

        public string UserCreatedBy { get; set; } = string.Empty;

        public int UserGroupId { get; set; }

        public DateTime? DateCompleted { get; set; }

        [JsonIgnore]
        public ICollection<ListItem> ListItems { get; set; } = new List<ListItem>();

    }
}

File: listr_app/ListR/lib/common/models/Data/shop_list.dart
import 'package:listr/common/models/Data/list_item.dart';

class ShopList {
  int? id;
  String? name;
  String? color;
  bool? isDeleted;
  bool? isCompleted;
  DateTime? dateCreated;
  DateTime? dateModified;
  String? userCreatedBy;
  int? userGroupId;
  DateTime? dateCompleted;
  List<ListItem>? listItems;

  ShopList(
      {this.id,
      this.name,
      this.color,
      this.isDeleted,
      this.isCompleted,
      this.dateCreated,
      this.dateModified,
      this.userCreatedBy,
      this.userGroupId,
      this.dateCompleted,
      this.listItems});

  ShopList.fromJson(Map<String, dynamic> json){
    id = json['id'];
    name = json['name'];
    color = json['color'];
    isDeleted = json['isDeleted'];
    isCompleted = json['isCompleted'];
    dateCreated = DateTime.tryParse(json['dateCreated']);
    dateModified = DateTime.tryParse(json['dateModified']);
    userCreatedBy = json['userCreatedBy'];
    userGroupId = json['userGroupId'];
    dateCompleted = DateTime.tryParse(json['dateCompleted']);
    
    listItems = <ListItem>[];
    for(var element in json['listItems']){
      listItems?.add(ListItem.fromJson(element));
    }
  }

  Map<String, dynamic> toMap(){
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['name'] = name;
    data['color'] = color ;
    data['isDeleted'] = isDeleted ;
    data['isCompleted'] = isCompleted ;
    data['dateCreated'] = dateCreated ;
    data['dateModified'] = dateModified;
    data['userCreatedBy'] = userCreatedBy ;
    data['userGroupId'] = userGroupId ;
    data['dateCompleted'] = dateCompleted ;
    data['listItems'] = listItems ;
    return data;
  }
}
